src/algebraic_geometry/projective_spectrum/topology.lean

```diff
@@ -0,0 +1,140 @@
+/-
+Copyright (c) 2020 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang, Johan Commelin
+-/
+
+import topology.opens
+import ring_theory.graded_algebra.homogeneous_ideal
+
+/-!
+# Projective spectrum of a graded ring
+
+The projective spectrum of a graded commutative ring is the subtype of all homogenous ideals that
+are prime and not containing all the irrelevant ideal.
+It is naturally endowed with a topology: the Zariski topology.
+
+## Notation
+- `R` is a commutative semiring;
+- `A` is a commutative ring and an `R`-algebra;
+- `𝒜 : ℕ → submodule R A` is the grading of `A`;
+
+## Main definitions
+
+* `projective_spectrum 𝒜`: The projective spectrum of a graded ring `A`, or equivalently, the set of
+  all homogeneous ideals of `A` that is both prime and relevant i.e. not containing all irrelevant
+  ideal. Henceforth, we call elements of projective spectrum relevant prime homogeneous ideals.
+* `zero_locus 𝒜 s`: The zero locus of a subset `s` of `A`
```

Comment:
By writing out the full name, it becomes a hyperlink in the generated documentation.
```suggestion
* `projective_spectrum.zero_locus 𝒜 s`: The zero locus of a subset `s` of `A`
```